tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/observability/metrics/WebSocketMetricsTestCase.java

```diff
@@ -0,0 +1,167 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.test.observability.metrics;
+
+import org.ballerinalang.jvm.observability.ObservabilityConstants;
+import org.ballerinalang.test.BaseTest;
+import org.ballerinalang.test.context.BServerInstance;
+import org.ballerinalang.test.util.websocket.client.WebSocketTestClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.Assert;
+import org.testng.annotations.AfterGroups;
+import org.testng.annotations.BeforeGroups;
+import org.testng.annotations.Test;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * Integration test for observability of metrics.
+ */
+@Test(groups = "websocket-metrics-test")
+public class WebSocketMetricsTestCase extends BaseTest {
+    private static BServerInstance serverInstance;
+    private static final Logger logger = LoggerFactory.getLogger(WebSocketMetricsTestCase.class);
+
+
+    private static final String RESOURCE_LOCATION = "src" + File.separator + "test" + File.separator +
+            "resources" + File.separator + "observability" + File.separator + "metrics" + File.separator;
+    private Map<String, String> expectedMetrics = new HashMap<>();
+    private static final ByteBuffer SENDING_BYTE_BUFFER = ByteBuffer.wrap(new byte[]{1, 2, 3, 4, 5});
+
+
+    @BeforeGroups(value = "websocket-metrics-test", alwaysRun = true)
+    private void setup() throws Exception {
+        serverInstance = new BServerInstance(balServer);
+        String balFile = new File(RESOURCE_LOCATION + "websocket-metrics-test.bal").getAbsolutePath();
+        List<String> args = new ArrayList<>();
+        args.add("--" + ObservabilityConstants.CONFIG_METRICS_ENABLED + "=true");
+        //args.add("--b7a.log.console.loglevel=INFO");
+        //args.add("&>");
+        //args.add("ballerina.log");
+        serverInstance.startServer(balFile, null, args.toArray(new String[args.size()]), new int[] { 9090 });
+        addMetrics();
+    }
+
+    @Test
+    public void testMetrics() throws Exception {
+        WebSocketTestClient client = new WebSocketTestClient("ws://localhost:9090/basic/ws");
+        client.handshake();
+        client.sendText("ds");
+        client.sendText("ds");
+        client.sendText("ds");
+        client.sendText("ds");
+        client.sendText("ds");
```

Comment:
We can have a local variable or a constant for the sending text instead of repeating it